cinema/serializers.py

```diff
@@ -68,13 +77,41 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available"
         )
 
 
-class MovieSessionDetailSerializer(MovieSessionSerializer):
+class TicketSerializer(serializers.ModelSerializer):
+    movie_session = MovieSessionListSerializer(read_only=True)
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+
+
+class TicketListSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("row", "seat")
+
+
+class MovieSessionDetailSerializer(serializers.ModelSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = TicketListSerializer(
+        source="tickets",
+        many=True,
+        read_only=True
+    )
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+
+class OrderListSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(read_only=True, many=True)
+
+    class Meta:
+        model = Order
+        fields = ("id", "tickets", "created_at")
```

Comment:
Why `OrderListSerializer` is used both for creation and listing? Read further about `Separation of concerns.
Also, at /orders, tickets, and orders themselves are created. Try to think how you can ensure `data consistency`, when creating an `Order` and its associated `Tickets`